src/main/java/jnr/posix/SpawnFileAction.java

```diff
@@ -47,7 +53,29 @@ public Open(String path, int fd, int flags, int mode) {
         }
 
         final boolean act(POSIX posix, Pointer nativeFileActions) {
-            return ((UnixLibC) posix.libc()).posix_spawn_file_actions_addopen(nativeFileActions, fd, path, flags, mode) == 0;
+            /*
+            This logic allocates a direct ByteBuffer to use for the path in order to work around systems that have not
+            patched CVE-2014-4043, in which older glibc versions do not make a defensive copy of the file path passed to
+            posix_spawn_file_actions_addopen. The buffer may be freed by the caller before it can be used in an
+            eventual posix_spawn call.
+
+            See https://bugzilla.redhat.com/show_bug.cgi?id=1983750 for a RHEL version of this issue.
+             */
+
+            // determine encoded byte array length
+            CharsetEncoder encoder = Charset.defaultCharset().newEncoder();
+            int bpc = (int) encoder.maxBytesPerChar();
+            int size = (path.length() + 1) * bpc;
+
+            // transcode to native buffer
+            nativePath = ByteBuffer.allocateDirect(size);
```

Comment:
I think this method can be called multiple times. Perhaps it's better to create and fill the buffer in the constructor?

Reply:
Agreed, I will make that change for this reason and to also avoid any race at `posix_spawn` call time.